Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -281,4 +282,138 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::symbol_match)
     return vm.call(*exec, rx, js_string(vm, s));
 }
 
+JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::symbol_replace)
+{
+    auto string_value = vm.argument(0);
+    auto replace_value = vm.argument(1);
+
+    // https://tc39.es/ecma262/#sec-regexp.prototype-@@replace
+    auto rx = regexp_object_from(vm, global_object);
+    if (!rx)
+        return {};
+    auto string = string_value.to_string(global_object);
+    if (vm.exception())
+        return {};
+
+    auto global_value = rx->get(vm.names.global).value_or(js_undefined());
+    if (vm.exception())
+        return {};
+
+    bool global = global_value.to_boolean();
+    if (global)
+        rx->regex().start_offset = 0;
+
+    // FIXME: Implement and use RegExpExec - https://tc39.es/ecma262/#sec-regexpexec
+    auto* exec = get_method(global_object, rx, vm.names.exec);
+    if (!exec)
+        return {};
+
+    Vector<Value> results;
```

Comment:
Let's make this a `Vector<Object&>` instead - the spec makes that assumption too (`Let result be ? RegExpExec(rx, S).` returns either object or null or throws). That way we also won't need to call `to_object()` on the value multiple times.

Reply:
It looks like `Vector` doesn't like holding references. Just declaring `Vector<Object&>` results in:
```
../.././AK/Vector.h: In instantiation of 'class AK::Vector<JS::Object&>':
../../Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp:311:21:   required from here
../.././AK/Vector.h:179:8: error: forming pointer to reference type 'JS::Object&'
  179 |     T* data()
```

Using `Vector<Object *>` instead seems to work fine. Is that safe here?

Reply:
Well, I was 100% convinced that's a thing, but you're right - it's not. This is fine :+1: